### PR TITLE
nonce impl

### PR DESCRIPTION
**Pull request by magiodev**
This PR implements a ProposalPayload struct as wrapper for proposal messages and a new nonce field which will then be persisted at state. New test_tube test has been added.

### CODE REVIEW

File: contracts/proposal/dao-proposal-single-instant/src/testing/test_tube.rs [hunk 1]
```diff
@@ -520,6 +549,185 @@ pub mod test_tube {
         );
     }
 
+    #[test]
+    #[ignore]
+    /// Test case of a proposal failing due to be replayed
+    fn test_dao_proposal_single_instant_ko_replay() {
```
Comment (0xLaurenzo) [inline contracts/proposal/dao-proposal-single-instant/src/testing/test_tube.rs:555]: nit: The exact logic of what this test tests and how it asserts that a replay happens is slightly hard to follow

**Review by 0xLaurenzo — APPROVED** [review]
lgtm, small nit around following the logic behind asserts

File: contracts/proposal/dao-proposal-single-instant/src/testing/test_tube.rs (hunk 1 above)
Comment (magiodev) [reply contracts/proposal/dao-proposal-single-instant/src/testing/test_tube.rs:555]: i removed unnecessary assertions around admin and treasury balances, as they are out of context and already tested out on the bank_send_ok test case. I also enhanced the comments around what the test is doing on the relevant lines. Let me know if thats fine!

**Review by 0xLaurenzo — APPROVED** [review]
lgtm
